invenio_curations/services/generators.py

```diff
@@ -57,3 +58,26 @@ def _condition(self, request=None, **kwargs):
             )
 
         return False
+
+
+class CurationModerators(Generator):
+    """Permission generator that allows users with the `moderation` role."""
+
+    def needs(self, **kwargs):
+        """Allow access for the moderation role."""
+        return [RoleNeed(current_curations.moderation_role_name)]
+
+
+class IfCurationRequestExists(ConditionalGenerator):
+    """Record-oriented generator checking if a curation request exists."""
+
+    def _condition(self, record=None, **kwargs):
+        """Check if the record has a curation request or not."""
+        if record is not None:
+            # We use the system identity here to avoid visibility issues
+            request = current_curations_service.get_review(
+                identity=system_identity, topic=record
+            )
+            return request is not None
+
+        return False
```

Comment:
permission generators that come in handy if operators decide to go down the route of overriding the record permission policy rather than creating access grants.
i'm using them in our own MR that makes config-tuw ready for the module: https://gitlab.tuwien.ac.at/fairdata/invenio-config-tuw/-/merge_requests/57

Reply:
Sure is nice to have this in the package!